contracts/modules/v2/Incidents.sol

```diff
@@ -0,0 +1,236 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-v4/token/ERC721/IERC721Receiver.sol";
+import "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
+
+import "../../interfaces/INXMToken.sol";
+import "../../interfaces/ITokenController.sol";
+import "../../interfaces/IMemberRoles.sol";
+import "../../interfaces/IPool.sol";
+import "../../interfaces/ICover.sol";
+import "../../interfaces/IAssessment.sol";
+import "../../interfaces/IIncidents.sol";
+
+import "../../abstract/MasterAwareV2.sol";
+
+/**
+ *  Provides a way for cover owners to submit claims and redeem the payouts and facilitates
+ *  assessment processes where members decide the outcome of the events that lead to potential
+ *  payouts.
+ */
+contract Incidents is IIncidents, IERC721Receiver, MasterAwareV2 {
+
+  // Ratios are defined between 0-10000 bps (i.e. double decimal precision percentage)
+  uint internal constant RATIO_BPS = 10000;
+
+  // Used in operations involving NXM tokens and divisions
+  uint internal constant PRECISION = 10 ** 18;
+
+  /* ========== STATE VARIABLES ========== */
+
+  INXMToken internal immutable nxm;
+
+  Configuration public override config;
+
+  Incident[] public override incidents;
+
+  /* ========== CONSTRUCTOR ========== */
+
+  constructor(address nxmAddress) {
+    nxm = INXMToken(nxmAddress);
+  }
+
+  function initialize(address masterAddress) external {
+    // The minimum cover premium per year is 2.6%. 20% of the cover premium is: 2.6% * 20% = 0.52%
+    config.rewardRatio = 52; // 0.52%
+    config.incidentExpectedPayoutRatio = 3000; // 30%
+    config.incidentPayoutDeductibleRatio = 9000; // 90%
+    master = INXMMaster(masterAddress);
+  }
+
+  /* ========== VIEWS ========== */
+
+  function memberRoles() internal view returns (IMemberRoles) {
+    return IMemberRoles(internalContracts[uint(IMasterAwareV2.ID.MR)]);
+  }
+
+  function assessment() internal view returns (IAssessment) {
+    return IAssessment(getInternalContractAddress(ID.AS));
+  }
+
+  function cover() internal view returns (ICover) {
+    return ICover(internalContracts[uint(IMasterAwareV2.ID.CO)]);
+  }
+
+  function getIncidentsCount() external override view returns (uint) {
+    return incidents.length;
+  }
+
+  /* === MUTATIVE FUNCTIONS ==== */
+
+  function submitIncident(
+    uint24 productId,
+    uint96 priceBefore,
+    uint32 date
+  ) external onlyAdvisoryBoard override {
+    uint96 activeCoverAmountInNXM = cover().activeCoverAmountInNXM(productId);
+
+    Incident memory incident = Incident(
+      0, // assessmentId
+      productId,
+      date,
+      priceBefore
+    );
+
+    uint expectedPayoutInNXM = activeCoverAmountInNXM * config.incidentExpectedPayoutRatio *
```

Comment:
make sure Anatol is aware he needs to provide this from a staking standpoint

Reply:
@shark0der @danoctavian I think capacity will be stored in Cover.sol, right?